ext/spl/spl_directory.c

```diff
@@ -2069,28 +2056,29 @@ PHP_METHOD(SplFileObject, __construct)
 		intern->u.file.open_mode_len = 1;
 	}
 
+	/* spl_filesystem_file_open() can generate E_WARNINGs which we want to promote to exceptions */
 	zend_replace_error_handling(EH_THROW, spl_ce_RuntimeException, &error_handling);
+	if (spl_filesystem_file_open(intern, use_include_path) == FAILURE) {
+		zend_restore_error_handling(&error_handling);
+		RETURN_THROWS();
+	}
+	zend_restore_error_handling(&error_handling);
```

Comment:
```suggestion
	zend_result retval = spl_filesystem_file_open(intern, use_include_path);
	zend_restore_error_handling(&error_handling);
	if (retval == FAILURE) {
		RETURN_THROWS();
	}
```
How I'd personally write this.